graphene_django/tests/test_query.py

```diff
@@ -431,6 +435,60 @@ class Query(graphene.ObjectType):
     assert result.data == expected
 
 
+@pytest.mark.skipif(not DJANGO_FILTER_INSTALLED,
+                    reason="django-filter should be installed")
+def test_should_query_node_filtering_with_distinct_queryset():
+    class FilmType(DjangoObjectType):
+
+        class Meta:
+            model = Film
+            interfaces = (Node, )
+            filter_fields = ('genre',)
+
+    class Query(graphene.ObjectType):
+        films = DjangoConnectionField(FilmType)
+
+        # def resolve_all_reporters_with_berlin_films(self, args, context, info):
```

Comment:
sorry last one :) Is this needed? 

I guess after resolving this @spockNinja or @syrusakbary can merge :)

Reply:
There might be a way to integrate it with the existing test suite but I didn't see a straightforward way to construct a query that would lead to one "half" of the resulting query being distinct and the other not. Doing it in this way keeps the addition localized to where it is needed. If you have a suggestion on how to integrate it with the existing model instead please let me know!